lib/actions/user.js

```diff
@@ -134,56 +158,116 @@ export function createOrUpdateUser (userData) {
+    const now = new Date()
+
+    // Request a new verification code if we are requesting a different number.
+    // or enough time has ellapsed since the last request (1 minute?).
+    // TODO: Should throttling be handled in the middleware?
+    if (number !== newPhoneNumber || timestamp + 60000 >= now) {
+      const { accessToken, apiBaseUrl, apiKey, loggedInUser } = getMiddlewareVariables(state)
+      const requestUrl = `${apiBaseUrl}${API_OTPUSER_PATH}/${loggedInUser.id}${API_OTPUSER_VERIFYSMS_PATH}/${encodeURIComponent(newPhoneNumber)}`
+
+      const { message, status } = await secureFetch(requestUrl, accessToken, apiKey, 'GET')
+
+      dispatch(setLastPhoneSmsRequestInfo({
+        number: newPhoneNumber,
+        status,
+        timestamp: now
+      }))
+
+      if (status === 'success') {
+        // Refetch user and update application state with new phone number and verification status.
+        // (This also refetches the user's monitored trip, and that's ok.)
+        await dispatch(fetchOrInitializeUser({ accessToken }))
       } else {
-        alert(`An error was encountered:\n${JSON.stringify(result)}`)
+        alert(`An error was encountered:\n${JSON.stringify(message)}`)
       }
     }
   }
 }
 
 /**
- * Deletes a logged-in user's monitored trip,
- * then, if that was successful, refreshes the redux monitoredTrips state.
+ * Validate the phone number verification code for the logged-in user.
  */
-export function deleteUserMonitoredTrip (id) {
+export function verifyPhoneNumber (code) {
   return async function (dispatch, getState) {
-    const { otp, user } = getState()
-    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+    const { accessToken, apiBaseUrl, apiKey, loggedInUser } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_OTPUSER_PATH}/${loggedInUser.id}${API_OTPUSER_VERIFYSMS_PATH}/${code}`
 
-    if (otpMiddleware) {
-      const { accessToken } = user
-      const deleteResult = await deleteTrip(otpMiddleware, accessToken, id)
+    const { data, message, status } = await secureFetch(requestUrl, accessToken, apiKey, 'POST')
 
-      if (deleteResult.status === 'success') {
-        // Reload user's monitored trips after deletion.
-        await dispatch(fetchUserMonitoredTrips(accessToken))
+    // If the check is successful, status in the returned data will be "approved".
+    if (status === 'success' && data) {
+      if (data.status === 'approved') {
+        // Refetch user and update application state with new phone number and verification status.
+        // (This also refetches the user's monitored trip, and that's ok.)
+        await dispatch(fetchOrInitializeUser({ accessToken }))
       } else {
-        alert(`An error was encountered:\n${JSON.stringify(deleteResult)}`)
+        // 'invalid' captures cases when users enter wrong/incorrect codes or expired codes.
+        alert('You entered in invalid code. Please try again.')
```

Comment:
Is there a way to distinguish whether the code has expired or if it is just invalid? I don't feel very strongly about it either way, but it'd be nice to communicate invalid codes due to expiration.

Reply:
Basically when the code expires, the backend returns a 500 error status, so that could be handled in the else block that deals with generic errors.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/cbf2d86940c5b4c51d9eb03969c22926c1a2ba7b.